scripts/galvo_phase_gain_calibration.py

```diff
@@ -0,0 +1,335 @@
+        info = [
+            # TODO add voltages when incorporated with AWG's
+            "E-beam scanner: {} V".format("N/A"),
+            "Galvo scanner: {} V".format("N/A"),
+            "Amplitude: {:.2f} um".format(PIXEL_SIZE_UM * numpy.abs(self.params[1])),
+        ]
+        ctx2 = cairo.Context(text_surface)
+        ctx2.set_source_rgb(1.00, 0.83, 0.00)
+        font_size = 20
+        ctx2.set_font_size(font_size)
+        ctx2.translate(font_size, font_size)
+        # Cairo doesn't do multiline text plotting, so loop over the text and show at a lower location.
+        for text in info:
+            ctx2.translate(0, font_size)
+            ctx2.show_text(text)
+            ctx2.stroke()
+
+        if wx.MAJOR_VERSION <= 3:
+            self.imageCtrl.SetBitmap(wx.BitmapFromImage(self.img))
+            self.imageCtrlSine.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(point_surface))
+            self.imageCtrlText.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(text_surface))
+        else:
+            self.imageCtrl.SetBitmap(wx.Bitmap(self.img))
+            self.imageCtrlSine.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(point_surface))
+            self.imageCtrlText.SetBitmap(wx.lib.wxcairo.BitmapFromImageSurface(text_surface))
+
+    def OnKey(self, event):
+        """Destroy the frame when the user presses q or Q."""
+        key = event.GetKeyCode()
+        if key in [ord("q"), ord("Q")]:
+            self.frame.Destroy()
+
+        # everything else we don't process
+        event.Skip()
+
+
+class ImagePasser(object):
+    """Pass an image in a Thread."""
+
+    def __init__(self):
+        self.image = None
+        self.available = threading.Event()
+        self.display = True
+
+
+def image_update(imp, window):
+    """
+    Update the image in the video displayer.
+
+    Parameters
+    ----------
+    imp: ImagePasser
+    window: VideoDisplayerSine
+
+    """
+    try:
+        while imp.display:
+            imp.available.wait()
+            imp.available.clear()
+            if not imp.display:
+                return
+            window.new_image(imp.image)
+    except Exception:
+        logging.exception("Failure during display")
+    finally:
+        logging.debug("Display thread ended")
+
+
+def live_display(ccd, dataflow, kill_ccd=True):
+    """
+    Acquire an image from one (or more) dataflow(s) and display it with a fitted sine wave overlay.
+
+    Parameters
+    ----------
+    ccd: odemis.model.DigitalCamera
+        A camera object.
+    dataflow: odemis.model.DataFlow
+        Dataflow object to access.
+    kill_ccd: bool
+        True if it is required to terminate the ccd after closing the window.
+    """
+    # create a window
+    window = VideoDisplayerSine("Live from %s.%s" % (ccd.role, "data"), ccd.resolution.value)
+    im_passer = ImagePasser()
+    t = threading.Thread(target=image_update, args=(im_passer, window))
+    t.daemon = True
+    t.start()
```

Comment:
It should be possible to integrate all of this into VideoDisplayerSine: it creates the thread, and all the attributes of "ImagePasser" are directly part of VideoDisplayerSine.
Then the new_image_wrapper just has to pass the image to a new method `VideoDisplayerSine.new_image_and_just_store_it_for_next_time_you_have_time_to_process_it()`. (or a better name ;-) )